File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    nums.sort()
    ponkemons = []
    for i in range(len(nums)):
        if i > 0 and nums[i] == nums[i-1]:
            continue
        if len(ponkemons) == len(nums) / 2:
            break
        ponkemons.append(nums[i])
    answer = len(ponkemons)
    return answer

File: Java/프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 1;
        Map<Integer, Integer> ponkemons = new HashMap<>();
        
        for (int num : nums) {
            ponkemons.put(num, ponkemons.getOrDefault(num, 0) + 1);
        }
        
        int maxKinds = ponkemons.size();
        int maxChoice = nums.length / 2;
        
        answer = Math.min(maxKinds, maxChoice);
        return answer;
    }
}